cmd/access/unstaked_access_node_builder.go

```diff
@@ -0,0 +1,125 @@
+package main
+
+import (
+	"strings"
+	"time"
+
+	"github.com/onflow/flow-go/cmd"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module"
+	"github.com/onflow/flow-go/module/local"
+	"github.com/onflow/flow-go/module/metrics"
+	"github.com/onflow/flow-go/network/p2p"
+	"github.com/onflow/flow-go/network/topology"
+)
+
+type UnstakedAccessNodeBuilder struct {
+	*FlowAccessNodeBuilder
+}
+
+func UnstakedAccessNode(anb *FlowAccessNodeBuilder) *UnstakedAccessNodeBuilder {
+	return &UnstakedAccessNodeBuilder{
+		FlowAccessNodeBuilder: anb,
+	}
+}
+
+func (uanb *UnstakedAccessNodeBuilder) Initialize() cmd.NodeBuilder {
+
+	uanb.validateParams()
+
+	uanb.enqueueUnstakedNetworkInit()
+
+	uanb.EnqueueMetricsServerInit()
+
+	uanb.RegisterBadgerMetrics()
+
+	uanb.EnqueueTracer()
+
+	uanb.PreInit(uanb.initUnstakedLocal())
+
+	return uanb
+}
+
+func (uanb *UnstakedAccessNodeBuilder) validateParams() {
+
+	logger := uanb.Logger()
+	// for an unstaked access node, the staked access node ID must be provided
+	if strings.TrimSpace(uanb.stakedAccessNodeIDHex) == "" {
+		logger.Fatal().Msg("staked access node ID not specified")
+	}
+
+	// and also the unstaked bind address
+	if uanb.unstakedNetworkBindAddr == cmd.NotSet {
+		logger.Fatal().Msg("unstaked bind address not set")
+	}
+}
+
+func (uanb *UnstakedAccessNodeBuilder) initUnstakedLocal() func(node cmd.NodeBuilder) {
+	return func(node cmd.NodeBuilder) {
+		// for an unstaked node, set the identity here explicitly since it will not be found in the protocol state
+		self := &flow.Identity{
+			NodeID:        node.NodeID(),
```

Comment:
It's really a shim at the moment, since the public NodeIDs (incl keys) should ultimately be found in the bootstrap files, but I agree a TODO comment marking how ephemeral the situation is would help.